algorithms/vi-map-helpers/src/vi-map-landmark-quality-evaluation.cc

```diff
@@ -91,4 +151,111 @@ void resetLandmarkQualityToUnknown(
   }
 }
 
+void findTracksOfInferiorDuplicateLandmarkObservations(
+    const vi_map::VIMap& map, const vi_map::MissionId& mission_id,
+    const vi_map::Landmark& landmark,
+    vi_map::TrackIdToKeypointsMap* track_ids_with_keypoints,
+    std::mutex* track_ids_with_keypoints_mutex) {
+  const vi_map::KeypointIdentifierList& landmark_observations =
+      landmark.getObservations();
+
+  std::unordered_map<
+      vi_map::VisualFrameIdentifier, vi_map::KeypointIdentifierList>
+      frames_with_observations;
+  vi_map::TrackIdToKeypointsMap track_id_to_observations;
+  vi_map::KeypointToTrackIdMap observation_to_track_id;
```

Comment:
A remaining issue here at the moment is the fact that an observation that was not tracked can be merged by loop closure. Meaning that it will have a track id of -1. This will create an unwanted grouping of observations removing way too many and/or forgetting some as they are later overwritten into `track_ids_with_keypoints` that the function modifier. We should still address this issue